_posts/2024-08-03-geospatial-techno.md

```diff
@@ -0,0 +1,120 @@
+
+[![](https://img.youtube.com/vi/3XCwZIJxi7M/0.jpg)](https://www.youtube.com/watch?v=3XCwZIJxi7M)
+
+## Introduction
+Filtering allows the selection of features that satisfy a specific set of conditions. Filters can be used in several contexts in GeoServer:
+- In WMS requests, select which features should be displayed on a map
+- In WFS requests, specify the features to be returned
+- In SLD documents, apply different symbolizations to features on a thematic map
+
+**Note.** This video was recorded on GeoServer 2.22.4, which is not the most up-to-date version. Currently, versions 2.24.x and 2.25.x are supported. To ensure you have the latest release, please visit this [link](https://geoserver.org/download/) and avoid using older versions of GeoServer.
+
+## Supported filter languages
+Data filtering in GeoServer follows the OGC Filter Encoding Specification, which provides a standard XML schema for encoding spatial, attribute, and temporal filters in GIS. This allows for customized queries to retrieve specific data from databases and web services while ensuring interoperability among GIS applications. GeoServer supports filters in both Filter Encoding Language and Common Query Language.
+
+### Filter Encoding Language
+The Filter Encoding language, defined by OGC standards, utilizes an XML-based syntax to select specific features, similar to the "WHERE" clause in SQL. A filter consists of a condition formed by Predicate elements and Logical operators, employing comparison and spatial operators to evaluate relationships between feature properties. In this session, we will explore various types of binary comparison operators, while the next sessions will cover spatial operators.
+
+### Common Query Language
+Common Query Language (CQL) is a language used in GeoServer for constructing filters and queries on geospatial data. It provides flexible and powerful options for filtering and retrieving specific subsets of data from GeoServer layers. In the upcoming sessions, we will dive into a detailed exploration of CQL, covering its various operations and practical usage.
+
+## Comparison operators
+These operators are part of Filter Encoding operators and are used in attribute-based queries to filter and retrieve specific features or data, based on their non-spatial attributes. The comparison operators include: binary comparison operators and value comparison operators.
+
+The binary comparison operators are:
+- PropertyIsEqualTo
+- PropertyIsNotEqualTo
+- PropertyIsLessThan
+- PropertyIsLessThanOrEqualTo
+- PropertyIsGreaterThan
+- PropertyIsGreaterThanOrEqualTo
+
+These operators contain two filter expressions to be compared. The first operand is often a `<PropertyName>`, but both operands may be any expression, function or literal value. Binary comparison operator elements may include an optional matchCase attribute, with the true or false value. The default value is true, but the comparisons do not check the case if the attribute has a false value.
```

Comment:
This may not be entirely correct, GeoServer is more forgiving than the OGC Standards here. CQL *requires* that the first operand be a PropertyName. 

ECQL is more relaxed and does the same thing as the XML Filter standard allowing the comparison of two Expressions.